File: 8vim/src/main/java/inc/flide/vim8/structures/yaml/Layer.java
package inc.flide.vim8.structures.yaml;

import com.fasterxml.jackson.annotation.JsonProperty;
import inc.flide.vim8.structures.Direction;
import java.util.Map;

public class Layer {
    @JsonProperty(required = true)
    public Map<Direction, Part> sectors;
}

File: 8vim/src/main/kotlin/inc/flide/vim8/ime/layout/models/yaml/versions/common/Layer.kt
package inc.flide.vim8.ime.layout.models.yaml.versions.common

import arrow.optics.optics
import com.fasterxml.jackson.annotation.JsonProperty
import inc.flide.vim8.ime.layout.models.Direction
import inc.flide.vim8.lib.ExcludeFromJacocoGeneratedReport
import java.util.EnumMap

@ExcludeFromJacocoGeneratedReport
@optics
data class Layer(
    @JsonProperty(required = true) val sectors: Map<Direction, Part> = EnumMap(
        Direction::class.java
    )
) {
    companion object
}
